### PR TITLE
Improve IRSA document for cross-account role for AWS Release v19

### PR DESCRIPTION
**Pull request by tuladhar**
Towards https://github.com/giantswarm/giantswarm/issues/29431

### What does this PR do?

- In AWS Release `v19.0.0`, we use `Cloudfront domain alias` for the identity provider URL. As such, for cross account roles we also need to mention new Provider URL needs to be created to use new `Cloudfront domain alias`. This need was [identity here](https://gigantic.slack.com/archives/C1R9481LZ/p1703059240250629?thread_ts=1703056503.494029&cid=C1R9481LZ). 

### What does it look like?

(If this is more than a textual change, a screenshot can help in some cases to speed up the review process.)

### Any background context you can provide?

(Please link public issues or summarize if not public.)

### Have you maintained the front matter?

(Please bump the last_review_date in case this qualifies as a review for an entire page. Provide user_questions which should be answered in the page. Provide a meaningful description.)

### CODE REVIEW

File: src/content/advanced/access-management/iam-roles-for-service-accounts/index.md [hunk 1]
```diff
@@ -495,6 +495,15 @@ Modify the trust entity of your AWS IAM roles with the new identity provider URL
 
 For cross account roles please follow the [guide](#cross-account) above and replace the `CLOUDFRONT_DOMAIN` with the new `CLOUDFRONT_ALTERNATE_DOMAIN`.
 
+Login into the AWS on the account where the cluster is running:
```
Comment (AndiDog) [inline src/content/advanced/access-management/iam-roles-for-service-accounts/index.md:498]:
```suggestion
Log in into AWS on the account where the cluster is running:
```

File: src/content/advanced/access-management/iam-roles-for-service-accounts/index.md [hunk 2]
```diff
@@ -495,6 +495,15 @@ Modify the trust entity of your AWS IAM roles with the new identity provider URL
...
+
+- Grab the URL of the Identity Provider in your current cluster `IAM > Identity Providers`. It will look like `https://irsa.CLUSTER_ID.k8s.INSTALLATION_NAME.BASE_DOMAIN`.
+
+Login into the account where the IAM role is located and create an Identity Provider in `IAM > Identity Providers`:
```
Comment (AndiDog) [inline src/content/advanced/access-management/iam-roles-for-service-accounts/index.md:502]:
```suggestion
Log into the account where the IAM role is located and create an Identity Provider in `IAM > Identity Providers`:
```

File: src/content/advanced/access-management/iam-roles-for-service-accounts/index.md (hunk 1 above)
Comment (AndiDog) [inline src/content/advanced/access-management/iam-roles-for-service-accounts/index.md:498]: This paragraph doesn't have an explanation. What is done here and why? Should this have a separate heading? Is it about making it work across accounts?

File: src/content/advanced/access-management/iam-roles-for-service-accounts/index.md [hunk 3]
```diff
@@ -495,6 +495,15 @@ Modify the trust entity of your AWS IAM roles with the new identity provider URL
...
+
+- Set `Provider URL` to the previously gathered URL and click the `Get thumbprint` to import the certificate.
```
Comment (AndiDog) [inline src/content/advanced/access-management/iam-roles-for-service-accounts/index.md:504]: Not sure if this is clear to customers. Should we add screenshots for the steps? Not really needed if this is a rare corner case.

File: src/content/advanced/access-management/iam-roles-for-service-accounts/index.md (hunk 1 above)
Comment (tuladhar) [reply src/content/advanced/access-management/iam-roles-for-service-accounts/index.md]: It is actually part of cross account roles mentioned above:
> For cross account roles please follow the [guide](#cross-account) above and replace the `CLOUDFRONT_DOMAIN` with the new `CLOUDFRONT_ALTERNATE_DOMAIN`.

Maybe, we should go with adding a separate heading.

Comment (tuladhar) [reply src/content/advanced/access-management/iam-roles-for-service-accounts/index.md]: I have updated the doc so, it doesn't relink to previous guide which might be confusing. Also, added screenshot.

**Pull request merged**